source/extensions/transport_sockets/tls/ssl_socket.h

```diff
@@ -118,7 +119,7 @@ class SslSocket : public Network::TransportSocket,
   // Ssl::PrivateKeyConnectionCallbacks
   void onPrivateKeyMethodComplete() override;
 
-  SSL* rawSslForTest() const { return ssl_; }
+  SSL* rawSsl() const { return info_->ssl_.get(); }
```

Comment:
To emphasize the appropriate usage of the `SSL*`, can you make `rawSsl()` private/protected, and keep the public `rawSslForTest()` (which can just be `return rawSsl();` please?